sdk/template/Azure.Template/src/Generated/Models/EvolvingModel.cs

```diff
@@ -40,6 +41,49 @@ internal EvolvingModel(int requiredInt, string requiredString, int? optionalInt,
             OptionalString = optionalString;
         }
 
+        /// <summary>
+        /// </summary>
+        /// <param name="targetVersion"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public bool IsValidInput(MultiVersionClientOptions.ServiceVersion targetVersion, out string message)
```

Comment:
Validation method that knows when different properties were added is generated as method on model.